workflow/artifacts/gcs/gcs.go

```diff
@@ -0,0 +1,129 @@
+package gcs
+
+import (
+	"cloud.google.com/go/storage"
+	"context"
+	"errors"
+	argoErrors "github.com/argoproj/argo/errors"
+	wfv1 "github.com/argoproj/argo/pkg/apis/workflow/v1alpha1"
+	log "github.com/sirupsen/logrus"
+	"io"
+	"os"
+)
+
+type GCSArtifactDriver struct {
+	Context context.Context
+}
+
+func (gcsDriver *GCSArtifactDriver) newGcsClient() (client *storage.Client, err error) {
+	gcsDriver.Context = context.Background()
+	client, err = storage.NewClient(gcsDriver.Context)
+	if err != nil {
+		return nil, argoErrors.InternalWrapError(err)
+	}
+	return
+
+}
+
+func (gcsDriver *GCSArtifactDriver) saveToFile(inputArtifact *wfv1.Artifact, filePath string) error {
+
+	log.Infof("Loading from GCS (gs://%s/%s) to %s",
+		inputArtifact.GCS.Bucket, inputArtifact.GCS.Key, filePath)
+
+	stat, err := os.Stat(filePath)
+	if err == nil {
```

Comment:
nit: for readability push up the first `err != nil` before this line, and then you don't need to test `err == nil` explicitly.

or use '&&' so it's a single conditional.

just a little awkward as-is w/ the nested block, and then `err != nil` below